appium/webdriver/webdriver.py

```diff
@@ -114,6 +114,113 @@ def _make_w3c_caps(caps: Dict) -> Dict[str, List[Dict[str, Any]]]:
+                def add_command(self):
+                    return ('GET', 'session/$sessionId/path/to/your/custom/url')
+
+        2. Creates a session with the extension.
+            # Appium capabilities
+            desired_caps = { ... }
+            driver = webdriver.Remote('http://localhost:4723/wd/hub', desired_caps,
+                extensions=[CustomYourCommand])
+
+        3. Calls the custom command
+            # Then, the driver calls a get request against
+            # `session/$sessionId/path/to/your/custom/url`. `$sessionId` will be
+            # replaced properly in the driver. Then, the method returns
+            # the `value` part of the response.
+            driver.custom_method_name()
+
+        4. Remove added commands (if needed)
+            # New commands are added by `setattr`. They remain in the module,
+            # so you should explicitly delete them to define the same name method
+            # with different arguments or process in the method.
+            driver.delete_extensions()
+
+
+        You can give arbitrary arguments for the command like the below.
+
+            class CustomYourCommand(ExtensionBase):
+                def method_name(self):
+                    return 'custom_method_name'
+
+                def test_command(self, argument):
+                    return self.execute(argument)['value']
+
+                def add_command(self):
+                    return ('post', 'session/$sessionId/path/to/your/custom/url')
+
+            driver = webdriver.Remote('http://localhost:4723/wd/hub', desired_caps,
+                extensions=[CustomYourCommand])
+
+            # Then, the driver sends a post request to `session/$sessionId/path/to/your/custom/url`
+            # with `{'dummy_arg': 'as a value'}` JSON body.
+            driver.custom_method_name({'dummy_arg': 'as a value'})
+
+
+        When you customize the URL dinamically with element id.
+
+            class CustomURLCommand(ExtensionBase):
+                def method_name(self):
+                    return 'custom_method_name'
+
+                def custom_method_name(self, element_id):
+                    return self.execute({'id': element_id})['value']
+
+                def add_command(self):
+                    return ('GET', 'session/$sessionId/path/to/your/custom/$id/url')
+
+            driver = webdriver.Remote('http://localhost:4723/wd/hub', desired_caps,
+                extensions=[CustomYourCommand])
+            element = driver.find_element_by_accessibility_id('id')
+
+            # Then, the driver calls a get request to `session/$sessionId/path/to/your/custom/$id/url`
+            # with replacing the `$id` with the given `element.id`
+            driver.custom_method_name(element.id)
+    """
+
+    def __init__(self, execute: Callable[[str, Dict], Dict[str, Any]]):
+        self._execute = execute
+
+    def execute(self, parameters: Any = {}) -> Any:
```

Comment:
Please do not set mutable objects as defaults: https://docs.quantifiedcode.com/python-anti-patterns/correctness/mutable_default_value_as_argument.html

Reply:
shall we use `*args, **kwargs` notation instead?

Reply:
Changed here as Dict explicitly to match with original execute: https://github.com/SeleniumHQ/selenium/blob/e46dba531896f67bdb93ba8d9e1d64f8b6b75b6f/py/selenium/webdriver/remote/webdriver.py#L380